skbio/alignment/_path.py

```diff
@@ -0,0 +1,305 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import numpy as np
+
+from skbio._base import SkbioObject
+from skbio.util._decorator import overrides, classonlymethod
+
+
+class AlignPath(SkbioObject):
+    def __init__(self, lengths, states, starts):
+        r"""Create an alignment path from segment lengths and states.
+
+        Parameters
+        ----------
+        lengths : array_like of int of shape (n_segments,)
+            Length of each segment in the alignment.
+        states : array_like of unit8 of shape (n_segments,) or (n_packs, n_segments)
+            Packed bits representing character (0) or gap (1) status per sequence per
+            segment in the alignment.
+        starts : array_like of int of shape (n_sequences,), optional
+            Start position (0-based) of each sequence in the alignment.
+
+        See Also
+        --------
+        skbio.sequence.Sequence
+        skbio.alignment.TabularMSA
+
+        """
+        # Number of sequences needs to be explicitly provided, because the packed bits
+        # does not contain this information. (It is merely in multiples of 8.)
+        self.lengths = np.asarray(lengths, dtype=int)
+        n_positions = self.lengths.sum()
+        self.states = np.atleast_2d(np.asarray(states, dtype=np.uint8))
+
+        # start positions
+        self.starts = np.asarray(starts, dtype=int)
```

Comment:
Is this okay as a regular `int`, or should we change this to `np.uint64` as well? As mentioned previously ~2Gbp is shorter than many genomes, and it is possible that the start position of a sequence will be beyond that point in the genome.

Reply:
Good point. I would suggest changing into `np.uint64` as well. Meanwhile, I don't think supporting negative start positions is necessary (I can't think of a use case).